packages/node/src/handlers.ts

```diff
@@ -391,11 +392,55 @@ export function requestHandler(
     const local = domain.create();
     local.add(req);
     local.add(res);
-    local.on('error', next);
+    local.on('error', err => {
+      // Request mode session capturing functionality is added here so that we are able to catch errors even if Sentry's
+      // error handler is not used
+      if (isAutosessionTrackingEnabled()) {
+        const scope = getCurrentHub().getStackTop().scope;
+        if (scope) {
+          /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+          const requestSessionStatus = (scope as any)._requestSessionStatus;
+          if (requestSessionStatus !== undefined) {
+            (scope as any)._requestSessionStatus = RequestSessionStatus.Errored;
+          }
+          /* eslint-enable @typescript-eslint/no-unsafe-member-access */
+        }
+      }
+      next(err);
+    });
+
     local.run(() => {
-      getCurrentHub().configureScope(scope =>
-        scope.addEventProcessor((event: Event) => parseRequest(event, req, options)),
-      );
+      const currentHub = getCurrentHub();
+      currentHub.configureScope(scope => {
+        scope.addEventProcessor((event: Event) => parseRequest(event, req, options));
+        if (isAutosessionTrackingEnabled()) {
+          const scope = currentHub.getScope();
+          if (scope) {
+            /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+
+            // Set `_requestSessionStatus` to Ok, at the beginning of the request and it is required to place it
+            // on the response as well because for some reason the scope is cleared in the `errorHandler`
+            // ToDo: Investigate why the scope does not propagate to the errorHandler
```

Comment:
We have no occurrences of `ToDo` or any other variation other than `TODO` in the code base, can we just use that?

Reply:
The comment here sounds a bit contradictory, "for some reason the scope is cleared in the `errorHandler`" vs "why the scope does not propagate to the errorHandler"

If it is cleared in `errorHandler` how can it not be propagated to it?